tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/environment/DockerImageExecutor.java

```diff
@@ -0,0 +1,199 @@
+        }
+
+        return containerId;
+    }
+
+    /**
+     *
+     * @param id Id of the environment. The container id.
+     * @param serviceName Service name in running environment. Can be null.
+     * @param script Script to execute
+     */
+    @Override
+    public void execScript(String id, String serviceName, String script) {
+        String execCreate = null;
+        try {
+            String sOrigAbsolutePath = TEST_SCRIPT_DIRECTORY + "/" + script;
+            String sNewAbsolutePath = "/opt/hawkular-apm-test-local";
+
+            /**
+             * Due to permissions problem with mounted volume script is moved to some local directory in container
+             *
+             * 1. create new directory in container
+             * 2. move there script
+             * 3. chmod script
+             * 4. execute script
+             */
+            String[] commands = new String[]{"bash", "-c",
+                    "mkdir " + sNewAbsolutePath + " && " +
+                    "cp " + sOrigAbsolutePath + " " + sNewAbsolutePath + " && " +
+                    "chmod +x " + sNewAbsolutePath + "/" + script  + " && " +
+                    sNewAbsolutePath + "/" + script};
+
+            execCreate = dockerClient.execCreate(id, commands,
+                    DockerClient.ExecCreateParam.attachStdout(), DockerClient.ExecCreateParam.attachStderr());
+
+            LogStream logStream = dockerClient.execStart(execCreate);
+//            System.out.println("\n\nTest script:\n" + logStream.readFully());
+        } catch (DockerException | InterruptedException ex) {
+            throw new EnvironmentException("Could not execute command", ex);
+        }
+    }
+
+    @Override
+    public void clean(String id) {
+        try {
+            dockerClient.killContainer(id);
+            dockerClient.removeContainer(id);
+        } catch (DockerException | InterruptedException ex) {
+            throw new EnvironmentException("Could not remove container: " + id, ex);
+        }
+    }
+
+    @Override
+    public void close() {
+        if (dockerClient != null) {
+            dockerClient.close();
+        }
+    }
+
+    private HostConfig hostConfig() {
+        /**
+         * Target directory is mounted to docker in order to access agent's jar
+         */
+        String hostOsMountDir = System.getProperties().getProperty("buildDirectory");
+
+        List<String> dockerInterfaceIpAddresses = InterfaceIpV4Address.getIpAddresses("docker0");
+        if (dockerInterfaceIpAddresses == null || dockerInterfaceIpAddresses.isEmpty()) {
+            throw new EnvironmentException("Could not find any ip address of network interface docker0");
+        }
+
+        String hostIpAddress = dockerInterfaceIpAddresses.iterator().next();
+
+        return HostConfig.builder()
+//                .networkMode("host")
+                .extraHosts("hawkular-apm:" + hostIpAddress)
```

Comment:
Ok, now I see why you need this to be a String :) I still think you should store the InetAddress on the map, and convert to the appropriate format at the time you consume the info.